frontend/packages/schema-editor/src/components/SchemaEditor/SchemaEditor.test.tsx

```diff
@@ -257,31 +253,101 @@ describe('SchemaEditor', () => {
     ).toBeDefined();
   });
 
-  test('close type when clicking on close button', async () => {
-    const selectedTypePointer = `#/${Keyword.Definitions}/TestType`;
-    const jsonSchema: JsonSchema = {
-      [Keyword.Properties]: {
-        someProp: { [Keyword.Type]: FieldType.String },
-        testProp: { [Keyword.Reference]: selectedTypePointer },
-      },
-      [Keyword.Definitions]: {
-        TestType: {
-          [Keyword.Type]: FieldType.Object,
-          [Keyword.Properties]: {
-            prop1: { [Keyword.Type]: FieldType.String },
-            prop2: { [Keyword.Type]: FieldType.String },
-          },
-        },
-      },
-    };
-    const schemaModel = SchemaModel.fromArray(buildUiSchema(jsonSchema));
+  it('close type when clicking on close button', async () => {
+    const setSelectedTypePointer = jest.fn();
+    const setSelectedNodePointer = jest.fn();
+    const schemaModel = SchemaModel.fromArray(buildUiSchema(jsonSchemaTypePanel));
+
     renderEditor({
-      appContextProps: { schemaModel, selectedTypePointer, setSelectedTypePointer: jest.fn() },
+      appContextProps: {
+        schemaModel,
+        selectedTypePointer,
+        setSelectedTypePointer,
+        setSelectedNodePointer,
+      },
     });
     const type = screen.getByTestId(testids.typeItem(selectedTypePointer));
     await act(() => user.click(type));
+    expect(
+      screen.getByText(textMock('schema_editor.types_editing', { type: 'TestType' })),
+    ).toBeInTheDocument();
     const closeType = screen.getByRole('button', { name: textMock('schema_editor.close_type') });
     await act(() => user.click(closeType));
-    expect(screen.queryByText(textMock('schema_editor.types_editing'))).toBeNull();
+    expect(setSelectedTypePointer).toHaveBeenCalledWith(null);
+    expect(setSelectedNodePointer).toHaveBeenCalledWith(undefined);
```

Comment:
Som hovedregel bør vi også validere at funksjonen bare har blitt kalt én gang. Det hjelper ikke at den har blitt kalt med `null` hvis den etterpå blir kalt med noe annet.
```suggestion
    expect(setSelectedTypePointer).toHaveBeenCalledTimes(1);
    expect(setSelectedTypePointer).toHaveBeenCalledWith(null);
    expect(setSelectedNodePointer).toHaveBeenCalledTimes(1);
    expect(setSelectedNodePointer).toHaveBeenCalledWith(undefined);
```
Gå gjerne over alle testene og se om det gir hensikt å gjøre tilsvarende sjekk der.

Reply:
Godt poeng, jeg har lagt ved slike sjekker nå. 
De gjentas ettersom testen trigger dem flere ganger, derfor har jeg lagt til flere sjekker underveis.
Jeg vurderte også å bruke `mockClear()` før siste action/klikk for å rydde opp. Denne løsningen virker ryddigere, men å ha sjekker ved hver action/klikk kan kanskje styrke testen? 🤔 
Foreløpig har jeg valgt å sjekke underveis, gjerne se hva du tenker er best når jeg pusher disse endringene :)

Reply:
Se bort i fra denne kommentaren min over - jeg har forenklet testene etter jeg såg litt nærmere på sammenhengene av testene og etter at jeg merget med main som ga en annerledes visning av elementene